plugins/codemirror/plugin.js

```diff
@@ -156,8 +166,8 @@
 		proto: {
 			setData: function (data) {
 				this.setValue(data);
-				this.status = 'ready';
-				this.editor.fire('dataReady');
+				this.value = 'ready';
```

Comment:
Why do we want to change this. (use `value` instead of `status`)

Our `codemirror` plugin is based on the `sourcearea` plugin from ckeditor, in which they use [`status`](https://github.com/ckeditor/ckeditor4/blob/7481c473b85b14f0bbfcd24fb5b368ebbc1fc3f9/plugins/sourcearea/plugin.js#L108) and not `value`.

Here we're creating a class based on [`CKEDITOR.editable`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editable.html) (i.e. a "subclass"), which does not have a `value` property, so that's why I'm asking.

Looking at the rest of the changes, I see that [here](https://github.com/liferay/liferay-ckeditor/pull/160/files#diff-9799baf72c5fc1f956bcb1e148ef99f646a31d84ca3559c4c19760b379853cf4R55) we're using 

```js
var oldData = instance.codeMirrorEditor.getValue();
```

But that's not the related to the `editable` subclass, and CodeMirror (or a CodeMirror instance to be more precise) will always have a [`getValue`](https://codemirror.net/doc/manual.html#getValue) method, so @NemethNorbert could you explain why this change is needed when you get a chance?

Reply:
Hey @NemethNorbert,

I checked these changes in `master` to check both [LPS-128735](https://issues.liferay.com/browse/LPS-128735) as well as [LPS-127948](https://issues.liferay.com/browse/LPS-127948); because the last time we touched our `codemirror` plugin it was to fix that issue. 

Everything works as expected so that's good news (in `master`, we get the **Expected Result** described in  [LPS-128735](https://issues.liferay.com/browse/LPS-128735)).

That said, I'm still convinced that we should use `this.status = 'ready';` and not `this.value = 'ready'`.

You can read more about an `editable`'s status [here](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editable.html#property-status). 

Unless we set that `status`, its value is going to be `"unloaded"`, as you can guess from the following screenshot 

![](https://user-images.githubusercontent.com/5572/114157381-df0cdf80-9923-11eb-9ef0-81931e0953d7.png)

If this needs to be merged quickly to prepare a release to be able to send it to `7.2.x`, `7.3.x` and `master`, I can send a new PR (based on this one, with the change I propose).

What do you think?